tests/fsharp/optimize/stringconcat/test.fs

```diff
@@ -0,0 +1,84 @@
+module Test.Test
```

Comment:
We normally put these tests in https://github.com/Microsoft/visualfsharp/tree/master/tests/fsharpqa/Source/CodeGen/EmittedIL

You can then use ..\\compareil.cmd to verify the il.

Reply:
I guess we could do that. But, I think it is ok to verify IL here too.